src/bio/bio_xstream.c

```diff
@@ -782,13 +782,23 @@ replace_bio_bdev(struct bio_bdev *old_dev, struct bio_bdev *new_dev)
 	}
 }
 
+bool prefix(const char *pre, const char *str)
+{
+	    return strncmp(pre, str, strlen(pre)) == 0;
+}
+
 int
 bdev_name2roles(const char *name)
 {
 	const char	*dst = strrchr(name, '_');
 	char		*ptr_parse_end = NULL;
 	unsigned	 int value;
 
+	if (prefix("HotInNvme", name)) {
```

Comment:
A better fix is to not parsing bdev name for hot plugged device:
- Remove the bdev_name2roles() call from create_bio_bdev() and add an extra parameter "roles" for this function;
- When the create_bio_bdev() is called from init_bio_bdevs(), call the bdev_name2roles() to get 'roles' first then pass it to create_bio_bdev();
- When the create_bio_bdev() is called from scan_bio_bdevs(), pass default 'roles' to create_bio_bdev();

Reply:
I just updated so that no roles get set by default if in PMem mode and "DATA" gets set when in MD-on-SSD mode, for consistency.